File: node-api/database/run_migration.js
const fs = require('fs');
const path = require('path');
const mysql = require('mysql2/promise');
const logger = require('../utils/logger');
const config = require('../config/database');

async function runMigration() {
    let connection;
    try {
        // Create connection
        connection = await mysql.createConnection(config);
        logger.info('Connected to database');

        // Read migration file
        const migrationPath = path.join(__dirname, 'migrations', '20240604_fix_documents_table.sql');
        const migration = fs.readFileSync(migrationPath, 'utf8');

        // Split migration into individual statements
        const statements = migration.split(';').filter(stmt => stmt.trim());

        // Execute each statement
        for (let statement of statements) {
            if (statement.trim()) {
                await connection.execute(statement);
                logger.info('Executed migration statement successfully');
            }
        }

        logger.info('Migration completed successfully');
    } catch (error) {
        logger.error('Error running migration:', error);
        throw error;
    } finally {
        if (connection) {
            await connection.end();
        }
    }
}

// Run migration if this file is run directly
if (require.main === module) {
    runMigration()
        .then(() => process.exit(0))
        .catch(error => {
            console.error('Migration failed:', error);
            process.exit(1);
        });
}

module.exports = { runMigration }; 

File: python-ml/migrations/run_migration.py
from app.utils.db import get_db_connection, safe_close_connection

def run_migration():
    conn = None
    cursor = None
    try:
        # Get connection from pool
        conn = get_db_connection()
        cursor = conn.cursor()
        
        # Read and execute migration file
        with open('migrations/002_fix_messages_table.sql', 'r') as f:
            # Split the SQL file into individual statements
            sql_statements = f.read().split(';')
            
            # Execute each statement
            for statement in sql_statements:
                if statement.strip():
                    try:
                        cursor.execute(statement + ';')
                        print(f"Executed: {statement[:50]}...")
                    except Exception as err:
                        print(f"Error executing statement: {err}")
                        print(f"Statement: {statement}")
            
        conn.commit()
        print("Migration completed successfully!")
        
    except Exception as err:
        print(f"Error: {err}")
    finally:
        safe_close_connection(conn, cursor)

if __name__ == '__main__':
    run_migration() 